### DISCUSSION TITLE
Correct way to add bootstrap or jquery-dependent custom js

### DISCUSSION THREAD

**Discussion by itsViney** [holger1411/understrap#1773] [Q&A] [answered]
I've very new to Understrap and I'm trying to make sure I don't pick-up any bad habits early on, so I'm wondering, what's the best way to add custom js that has a dependancy such as jquery or bootstrap? Simply writing jquery or referencing bootstrap in custom-javascript.js doesn't work since it seems to be loaded before jquery and bootstrap's js. How _should_ I be doing this? For example:

`var scrollSpy = new bootstrap.ScrollSpy(document.body, {
target: '#jumpmenu-content'
})`

**Comment by IanDelMar** [comment, marked as answer]
@itsViney are there any js errors reported in the console? The contents of custom-javascript.js are loaded after jquery and bootstrap.

I guess this is related to #1583.

**Reply by bacoords** [reply]
Correct - I think you can do 

`var scrollSpy = new ScrollSpy(document.body, { target: '#jumpmenu-content' })`

without the bootstrap name

**Reply by itsViney** [reply]
Thanks for the reply. I've tried removing the "bootrap." name as you suggested but I still get the same problem and a console error as I was before:

**Uncaught ReferenceError: ScrollSpy is not defined**

**Reply by IanDelMar** [reply]
> Thanks for the reply. I've tried removing the "bootrap." name as you suggested but I still get the same problem and a console error as I was before:
> 
> **Uncaught ReferenceError: ScrollSpy is not defined**

You have to import scrollspy first. See https://github.com/understrap/understrap/issues/1583#issuecomment-1022444176

**Reply by itsViney** [reply]
Ahh, thanks so much @IanDelMar ! I'd read that answer yesterday but must have glazed over the crucial **import** line. 🤦‍♂️

For anyone stumbling across this later, this is my corrected code:

```
import scrollSpy from 'bootstrap/js/dist/scrollspy'
var scrollSpycontainer = new scrollSpy(document.body, { target: '#jumpmenu-content' });
```
